src/extension.ts

```diff
@@ -103,7 +103,7 @@ export class Extension implements RunHooks {
     });
 
     this._settingsModel = new SettingsModel(vscode, context);
-    this._reusedBrowser = new ReusedBrowser(this._vscode, this._settingsModel, this._envProvider.bind(this));
+    this._reusedBrowser = new ReusedBrowser(this._vscode, this._settingsModel, this._envProvider.bind(this), this._showBrowserForRecording.bind(this));
```

Comment:
Instead of passing this as a callback, can we create a file and run the test before calling `ReusedBrowser.record()`? This will make dependencies flow forward instead of both ways.